### DISCUSSION TITLE
Custom Library not found

### DISCUSSION THREAD

**Discussion by Heizelmann** [arendst/Tasmota#17781] [Q&A] [answered]
I am new here and try to implement my first custom I2C Driver. Basically ist works find but I am not sure how to place a custom library. I tried to simply put it in a new subfolder of lib/lib_i2c. But my driver which is located in a subdirectory in tasmota/tasmota_xdrv_driver couldn't find it. The library is called **MCP47x6** and the driver uses **include "MCP47x6"** The Error message is 

/workspace/Tasmota/tasmota/tasmota_xdrv_driver/xdrv_61_dimmer_I2C.ino:25:21: fatal error: MCP47X6.h: No such file or directory

*****************************************************************
* Looking for MCP47X6.h dependency? Check our library registry!
*
* CLI  > platformio lib search "header:MCP47X6.h"
* Web  > https://registry.platformio.org/search?q=header:MCP47X6.h
*
*****************************************************************

 #include  "MCP47X6.h" //
                     ^
compilation terminated.

I also made these changes

In my_user_config.h:
//#define USE_DS3502                                 // [I2CDriver67] Enable DS3502 digital potentiometer (I2C address 0x28 - 0x2B) (+0k4 code)
#define USE_MCP47X6                                 // [I2CDriver67] Enable MCP47x6 PCM device (I2C address 0x30 )

In tasmota_configurations.h
//#define USE_DS3502                             // [I2CDriver67] Enable DS3502 digital potentiometer (I2C address 0x28 - 0x2B) (+0k4 code)
#define USE_MCP47X6                             // [I2CDriver67] Enable USE_MCP47X6 digital pwm (I2C address 0x30) ( code)

**Comment by barbudor** [comment, marked as answer]
What you describe seems correct but without seeing that in details for myself, it's difficult to estimate where is your problem
It doesn't looks like your fork on github is up to date as it is missing the xdrv_61_dimmer_I2C.ino: file
If you update your fork, then it could be possible for us to check exactly what is wrong (I can do that tonight)

**Reply by Heizelmann** [reply]
Sory, Push is done right now.

**Reply by barbudor** [reply]
I'll have a look after diner

**Reply by barbudor** [reply]
``` 
Environment    Status    Duration
-------------  --------  ------------
tasmota        SUCCESS   00:00:26.904
```

The right location is in `lib/lib_i2c` 
Remove the one in `tasmota/` 

The error was your `library.properties` file which states `architectures=avr` thus limiting this library only for Atmel AVR MCU
Replace the `avr` by a `*` and it compiles straight away

**Comment by Heizelmann** [comment]
Merci beaucoup - Thanks a lot for the quick help!

**Reply by barbudor** [reply]
Gern geschehen
